cli/azd/pkg/account/manager.go

```diff
@@ -182,11 +182,21 @@ func (m *Manager) SetDefaultLocation(ctx context.Context, subscriptionId string,
 }
 
 // Checks whether account related defaults of subscription and location have previously been set
-func (m *Manager) HasDefaults() bool {
-	_, hasDefaultSubscription := m.config.Get(defaultSubscriptionKeyPath)
+func (m *Manager) HasDefaults(ctx context.Context) (hasDefaults bool, areDefaultAccessible bool) {
+	sub, hasDefaultSubscription := m.config.Get(defaultSubscriptionKeyPath)
 	_, hasDefaultLocation := m.config.Get(defaultLocationKeyPath)
+	areDefaultAccessible = true
 
-	return hasDefaultSubscription && hasDefaultLocation
+	// check if the defaults are valid/accessible
+	if hasDefaultSubscription {
+		_, err := m.azCli.GetAccount(ctx, fmt.Sprint(sub))
```

Comment:
We now have two places (here and in getDefaultSubscription) where we have to check to see if the stored value can be used.  I'm wondering if there's some refactoring we can do to remove the logic.

Part of me wonders what things would look like if we keep the signature of HasDefaults here and then had `getDefaultSubscription` return something like `ErrNoSubscription` in cases where we fail to discover information about the saved subscription (i.e. the call falls with a 404 error).

The callers of `getDefaultSubscription` could then decide how to make the policy decision on what they wanted to do instead of that behavior being hardcoded in to `getDefaultSubscription`.  I'm not sure if that would end up being cleaner overall.

Reply:
My first approach was to save the `invalid defaults` state within the account manager.  But it turns out that the manager can be created multiple times from multiple other components (not a single instance per azd execution). So it can't hold state.

I considered then using the azd environment to hold this info, of the exe environment, but I ended up doing the double check to avoid depending making the account manager dependent on external sources.

I would prefer if we could make this manager similar to the `console` and have it injected and passed through, or registered in the ioc. Any approach to keep it stateful. But for now, the logic is kept within the account manager only

Reply:
I updated the implementation to use the already registered account.Manager from the global ioc,  this way, we can keep the state of the account in memory